lgc/patch/PatchEntryPointMutate.cpp

```diff
@@ -950,8 +951,15 @@ void PatchEntryPointMutate::setFuncAttrs(Function *entryPoint) {
 
   // NOTE: Remove "readnone" attribute for entry-point. If GS is empty, this attribute will allow
   // LLVM optimization to remove sendmsg(GS_DONE). It is unexpected.
+#if LLVM_MAIN_REVISION && LLVM_MAIN_REVISION < 440919
+  // Old version of the code
   if (entryPoint->hasFnAttribute(Attribute::ReadNone))
     entryPoint->removeFnAttr(Attribute::ReadNone);
+#else
+  // New version of the code (also handles unknown version, which we treat as
+  // latest)
+  entryPoint->setMemoryEffects(MemoryEffects::unknown());
```

Comment:
@amdrexu does this look OK? I don't have a good understanding of this code and there is no lit test for it.

Reply:
Can you remove the REVISION check here too - or is this one required? (Based on Nicolai's comments in #2119 )
If it is, then I think that the revision number I used there was wrong. It probably requires calculating again.

Reply:
I couldn't find an easy way to do this without an LLVM_MAIN_REVISION check. APIs like `func->setOnlyReadsMemory()` generally only let you add more restrictions to the function's memory effects. There does not seem to be an easy way to remove restrictions.

I did double check this revision number, and it corresponds exactly to the relevant upstream commit.

Reply:
Ok - thanks - I thought you'd re-used the one I did in #2119 - but I see now they're slightly different?

Reply:
Yes it is intentionally slightly different, 440919 instead of 440909.

Reply:
Sorry - should not have marked this as resolved.